### ISSUE TITLE
from_array does not take masked arrays into account

### ISSUE THREAD

**Issue by adehecq** [GlacioHack/GeoUtils#149]
**Describe the bug**
if `my_array` is a np.ma.masked_array, calling
`out_raster = gu.Raster.from_array(my_array, transform, crs)`
will not preserve my_array's mask.

**To Reproduce**
```
import geoutils as gu
img = gu.Raster(gu.datasets.get_path('landsat_B4'))
img.set_ndv(255)
print(img.data[0, 0, 0])
```
-> print `masked`

```
out_raster = gu.Raster.from_array(img.data, img.transform, img.crs)
print(out_raster.data[0, 0, 0])
```
-> print 255

**Expected behavior**
The mask of img.data should be preserved and second print statement returns a masked value.

**System (please complete the following information):**
irrelevant

**Comment by adehecq** [comment]
The above example is incomplete, as it would be solved by calling:
`out_raster = gu.Raster.from_array(img.data, img.transform, img.crs, nodata=img.nodata)`

The issue arise when several nodata values exist, for example when differencing two rasters with different no data:
```
import geoutils as gu
img = gu.Raster(gu.datasets.get_path('landsat_B4'))
img.set_ndv(255)
img2 = gu.Raster(gu.datasets.get_path('landsat_B4'))
img2.set_ndv(254)
print(img.data[0, 0, 0])
print(img2.data[0, 0, 592])
```
-> print two masked values
```
diff = img.data - img2.data
out_raster = gu.Raster.from_array(diff, img.transform, img.crs, nodata=img.nodata)
print(out_raster.data[0, 0, 0])
print(out_raster.data[0, 0, 592])
```
print -- and 254.

**Expected behavior**
Second print should return two masked values.

**Comment by adehecq** [comment]
Solved by #150
